tensorflow/core/graph/mkl_layout_pass_test.cc

```diff
@@ -1547,288 +1562,345 @@ TEST_F(MklLayoutPassTest, NodeRewrite_Dequantize_Negative_Non_SCALED_Mode) {
-      "node { name: 'C' op: 'DoubleInput'}"
-      "node { name: 'D' op: '_FusedConv2D'"
-      " attr { key: 'T'                value { type: DT_DOUBLE } }"
-      " attr { key: 'num_args'         value { i: 1 } }"
-      " attr { key: 'data_format'      value { s: 'NCHW' } }"
-      " attr { key: 'strides'          value { list: {i: 1, i:1, i:1, i:1} } }"
-      " attr { key: 'padding'          value { s: 'SAME' } }"
-      " attr { key: 'dilations'        value { list: {i: 1, i:1, i:1, i:1} } }"
-      " attr { key: 'fused_ops'        value { list: {s: 'BiasAdd'} } }"
-      " attr { key: 'epsilon'          value { f: 0.001 }}"
-      " input: ['A', 'B', 'C']}"
-      "node { name: 'E' op: 'Zeta' attr { key: 'T' value { type: DT_DOUBLE } }"
-      " input: ['D', 'C'] }");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(DoubleInput);B(DoubleInput);C(DoubleInput);"
-            "D(_FusedConv2D);E(Zeta)|A->D;B->D:1;C->D:2;C->E:1;D->E");
-}
+#define REGISTER_TEST(NAME, T, INPUT)                                          \
+  TEST_F(MklLayoutPassTest, NAME##_##T) {                                      \
+    InitGraph(                                                                 \
+        "node { name: 'A' op: 'DoubleInput'}"                                  \
+        "node { name: 'B' op: 'DoubleInput'}"                                  \
+        "node { name: 'C' op: 'DoubleInput'}"                                  \
+        "node { name: 'D' op: '_FusedConv2D'"                                  \
+        " attr { key: 'T'                value { type: DT_DOUBLE } }"          \
+        " attr { key: 'num_args'         value { i: 1 } }"                     \
+        " attr { key: 'data_format'      value { s: 'NCHW' } }"                \
+        " attr { key: 'strides'          value { list: {i: 1, i:1, i:1, i:1} " \
+        "} }"                                                                  \
+        " attr { key: 'padding'          value { s: 'SAME' } }"                \
+        " attr { key: 'dilations'        value { list: {i: 1, i:1, i:1, i:1} " \
+        "} }"                                                                  \
+        " attr { key: 'fused_ops'        value { list: {s: 'BiasAdd'} } }"     \
+        " attr { key: 'epsilon'          value { f: 0.001 }}"                  \
+        " input: ['A', 'B', 'C']}"                                             \
+        "node { name: 'E' op: 'Zeta'"                                          \
+        "attr { key: 'T' value { type: DT_DOUBLE } }"                          \
+        " input: ['D', 'C'] }");                                               \
+    EXPECT_EQ(DoMklLayoutOptimizationPass(),                                   \
+              "A(DoubleInput);B(DoubleInput);C(DoubleInput);"                  \
+              "D(_FusedConv2D);E(Zeta)|A->D;B->D:1;C->D:2;C->E:1;D->E");       \
+  }
+REGISTER_TEST_ALL_TYPES(NodeRewrite_FusedConv2D_Negative2);
```

Comment:
Do we need to wrap this in `REGISTER_TEST`? We only want to test with `DT_DOUBLE` anyway.

Reply:
Thanks for the review, Penporn. I can remove this. Just keeping it that way to make it consistent with others, and in case, we need to update the test for other types besides Double. If you prefer though, I can remove REGISTER_TEST macro.

Reply:
Then how about we call `REGISTER_TEST(NodeRewrite_FusedConv2D_Negative2, DT_DOUBLE, DoubleInput);` for now, just to make it clear that this has nothing to do with `float32` and `bfloat16`?